### PR TITLE
refactor: Migrate audio-asset to react

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/components/asset/AudioAsset.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import {AssetRepository} from '../../assets/AssetRepository';
+import {registerReactComponent} from 'Util/ComponentUtil';
+
+const logger = getLogger('AudioAssetComponent');
+
+export interface AudioAssetProps {
+  assetRepository?: AssetRepository;
+  className?: string;
+  header: boolean;
+  // Does the asset have a visible header?
```
Comment (Yserz) [inline src/script/components/asset/AudioAsset.tsx]:
```suggestion
  /* Does the asset have a visible header? */
```
and should be above the `header` attribute

File: src/script/components/asset/AudioAsset.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  header,
+  message,
+  className,
+  teamState = container.resolve(TeamState),
+  assetRepository = container.resolve(AssetRepository),
+}) => {
+  const asset = message.getFirstAsset() as FileAsset;
+  const [audioElement, setAudioElement] = useEffectRef<HTMLAudioElement>();
+  const isFileSharingReceivingEnabled = teamState.isFileSharingReceivingEnabled();
+  const {transferState, uploadProgress, cancelUpload} = useAssetTransfer(message);
```
Comment (Yserz) [inline src/script/components/asset/AudioAsset.tsx]: `useAssetTransfer` returns a function `loadAsset` which can be used in line 78. This way we don't need to inject the assetRepository anymore.

File: src/script/components/asset/AudioAsset.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    <div className={cx('audio-asset', className)} data-uie-name="audio-asset" data-uie-value={asset.file_name}>
+      <audio ref={setAudioElement} src={audioSrc} onTimeUpdate={onTimeupdate} />
+      {!message.isObfuscated() && (
+        <>
+          {header && (
+            <div style={{width: '100%'}}>
+              <AssetHeader message={message} />
+            </div>
+          )}
+          {!isFileSharingReceivingEnabled && <RestrictedAudio />}
```
Comment (Yserz) [inline src/script/components/asset/AudioAsset.tsx]: Could be a ternary operator

File: src/script/components/asset/AudioAsset.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    transferState={transferState}
+                    uploadProgress={uploadProgress}
+                  />
+
+                  {transferState !== AssetTransferState.UPLOADING && (
+                    <>
+                      <span className="audio-controls-time label-xs" data-uie-name="status-audio-time">
+                        {formatSeconds(audioTime)}
+                      </span>
+                      {showLoudnessPreview && (
```
Comment (Yserz) [inline src/script/components/asset/AudioAsset.tsx]: Could be a ternary operator

File: src/script/components/asset/AudioAsset.tsx (hunk 2 above)
Comment (thisisamir98) [reply src/script/components/asset/AudioAsset.tsx]: https://github.com/wireapp/wire-webapp/pull/11577/commits/fbf7479538dc1209f6457934b3b5de970eef0b07

File: src/script/components/asset/AudioAsset.tsx (hunk 4 above)
Comment (thisisamir98) [reply src/script/components/asset/AudioAsset.tsx]: https://github.com/wireapp/wire-webapp/pull/11577/commits/bee43b1c824461a571125a4848147818f55e919e

File: src/script/components/asset/AudioAsset.tsx (hunk 3 above)
Comment (thisisamir98) [reply src/script/components/asset/AudioAsset.tsx]: https://github.com/wireapp/wire-webapp/pull/11577/commits/bee43b1c824461a571125a4848147818f55e919e

**Review by ffflorian — APPROVED** [review]

**Review by ffflorian — APPROVED** [review]

**Review by Yserz — APPROVED** [review]

**Pull request merged**